### DISCUSSION TITLE
Best practise to wait and redirect to a page in NextJS 13.4.4

### DISCUSSION THREAD

**Discussion by nadunperera** [vercel/next.js#50837] [Help]
### Summary

I have the below function in my ```OnboardingForm``` component at ```/onboarding``` page.

```
  async function handleFormData(formData: FormData) {
    const result = await createUserFromForm(formData, clerkUserId as string)

    if (result) {
      await setUserDataAtClerk(
        clerkUserId as string,
        formData.get('firstName') as string,
        formData.get('lastName') as string
      )

      redirect('/dashboard')
    } else {
      // Add error handling logic when createUserFromForm doesn't return true
      console.error('User could not be created from form data')
    }
  }
```

Problem:

 1. The ```setUserDataAtClerk``` function updates the user's ```isActive``` status
    to true. This operation takes some time to complete and propagate to
    the auth service. 
 2. I have an ```authMiddleware``` checks for the ```isActive``` status
    and redirects users who aren't active to ```/onboarding```. 
 3. After calling ```setUserDataAtClerk```, my code tries to redirect the user to
    ```/dashboard```. However, at that point, the update to ```isActive``` is not
    propagated, and the middleware still see the old
    status and redirect back to ```/onboarding```.

I am using next.js 13.4.4. 

What is the best way to overcome this challenge?

**Comment by bartekczyz** [comment]
I have the same problem. After submitting the form, I try to redirect the user to another page (behind auth guard using middleware), but it won't run. Deactivating the middleware helps. Oh, also, it's only broken on a production build, it works fine on dev :/
